src/main/java/org/apache/datasketches/hll/BaseHllSketch.java

```diff
@@ -118,8 +120,14 @@ public static final int getSerializationVersion(final Memory mem) {
    * <a href="{@docRoot}/resources/dictionary.html#numStdDev">Number of Standard Deviations</a>
    * @return the current (approximate) RelativeError
    */
-  public double getRelErr(final boolean upperBound, final boolean unioned,
+  public static double getRelErr(final boolean upperBound, final boolean unioned,
       final int lgConfigK, final int numStdDev) {
+    HllUtil.checkLgK(lgConfigK);
+    if (lgConfigK > 12) {
+      final double rseFactor = unioned ? HLL_NON_HIP_RSE_FACTOR : HLL_HIP_RSE_FACTOR;
+      final int configK = 1 << lgConfigK;
+      return (upperBound ? -1.0 : 1.0) * (numStdDev * rseFactor) / Math.sqrt(configK);
```

Comment:
If we know that there's a guaranteed direction for an error, it's reasonable and I'd argue correct to reflect that.

Reply:
this method did return negative values for UB before. It did not handle lgK > 12. I fixed that.
I agree that returning negative is confusing, but changing that might break somebody's code. Perhaps it is fine now in Java since we are getting ready for a major release, but I am no sure how to approach this in C++, where we had a major release recently.